src/main/java/org/commcare/cases/entity/NodeEntityFactory.java

```diff
@@ -199,10 +199,11 @@ public final boolean isEntitySetReady() {
      * Caches the provided entities. Default implementation throws RuntimeException.
      * Subclasses should override this method if they support caching.
      *
-     * @param entities List of entities to cache
+     * @param entities     List of entities to cache
+     * @param skipLazyLoad true if we want to calculate even lazy-loading properties as part of the process
      * @throws RuntimeException if caching is not supported
      */
-    public void cacheEntities(List<Entity<TreeReference>> entities) {
+    public void cacheEntities(List<Entity<TreeReference>> entities, Boolean skipLazyLoad) {

```

Comment:
Probably a nitpick, but it feels a bit to me like the "force this to be synchronous" call should normally either be a separate named function, or potentially part of the constructor (or a 'set' function) / nature of the factory. I think it's a little funny for individual method signatures to be either synchronous and blocking or asynchronous based on an arg. 

Reading the implementations it does look like we might already know whether we want/need it to be blocking when it's created / before this method is called.

Reply:
https://github.com/dimagi/commcare-core/pull/1454/commits/61d0a096f4679bc09d16739bf4f55fc03a4fb4ab